includes/benchmark.py

```
@@ -104,50 +114,186 @@ def run_test(cfg, level, skipverify):
+
+    return compsize,comptime,decompsize,decomptime,hashfail
+
+def prepare_singlemode(testconfig, cfgSingle):
+    ''' Single mode, we use the same file for every level, but support separate files for compress/decompress '''
+    cfg = util.dict_to_namedt(testconfig)
+    separate_files = False
+    tempfiles = {level: {} for level in cfg.levels}
+
+    # Same input file for compress + decompress?
+    same_input_file = False
+    if cfg.do_compress and cfgGenDecomp.get('srcFile') == cfgGenComp.get('srcFile'):
```

Ruff (0.14.14)

[error] Undefined name `cfgGenDecomp`

(F821)